### ISSUE TITLE
large history leads to error 500 from api due to cut off json in database

### ISSUE THREAD

**Issue by tldev-de** [hedgedoc/hedgedoc#2623]
### Description

`/history` api throws error 500 due to cut off json in database

### Steps to reproduce

1. you need to have a large history, e.g. I have currently about 35 pages history in hedgedoc
2. open the front page of hedgedoc
3. `/history` throws an error 500, no history is shown
4. if `/history` does not throw an error 500, you need to open more notes

### Expected behaviour

`/history` api returns history data with status code 200

### Logs

```
2022-09-01T07:11:33.131Z info: 	deserializeUser: 563c5c12-XXXX-XXXX-XXXX-4b08ac8b11ec
2022-09-01T07:11:33.134Z error: 	read history failed: SyntaxError: Unexpected end of JSON input
2022-09-01T07:11:33.137Z info: 	10.75.4.2 - - [01/Sep/2022:07:11:33 +0000] "GET /history HTTP/1.1" 500 - "https://notes.XXXX.de/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36"
```

### Config

```
version: '2'
services:
  app:
    image: quay.io/hedgedoc/hedgedoc:1.9.3
    environment:
      - "CMD_DOMAIN=notes.XXXXXX.de"
      - "CMD_URL_ADDPORT=false"
      - "CMD_PROTOCOL_USESSL=true"
      - "CMD_DB_URL=mysql://XXXXX:XXXXXXXXXXX@mysql.XXX.XXXX.de:3306/hackmd"
      - "CMD_DEFAULT_PERMISSION=editable"
      - "CMD_EMAIL=false"
      - "CMD_IMAGE_UPLOAD_TYPE=filesystem"
      - "CMD_SESSION_SECRET=XXXXXXX"
    volumes:
      - uploads:/hedgedoc/public/uploads
    restart: always
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.notes.rule=Host(`notes.XXXX.XXXXX.de`)"
      - "traefik.http.services.notes.loadbalancer.server.port=3000"
volumes:
  uploads:
```

### Your Setup

- Host OS: Debian 10 with Docker Container `quay.io/hedgedoc/hedgedoc:1.9.3`
- NodeJS version: v16.16.0
- HedgeDoc version: 1.9.3

### Additional context

In the mysql database, the `Users`.`history` column is of type text, which limits it to 65,535 chars. When reaching this number of chars, the text is cut off, which results in invalid json and therefore the problem here. My quick fix was to use `longtext` instead of `text` and repairing the json syntax manually.

**Comment by mrdrogdrog** [comment]
I created a quick migration that changes the type of the column. It seems like the problem occurs only in mysql because sqlite and postgres use a variable length for text.

**Comment by SISheogorath** [comment]
Well, I guess we missed this back then:

 https://github.com/hedgedoc/hedgedoc/blob/master/lib/migrations/20171009121200-longtext-for-mysql.js

**Comment by mrdrogdrog** [comment]
Fixed with #2624 . Will be included in the next release and is also available in the current nightly image.
